go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -468,6 +469,7 @@ func (tm *TabletManager) checkMastership(ctx context.Context, si *topo.ShardInfo
 			})
 		case err == nil:
 			if oldTablet.Type == topodatapb.TabletType_MASTER {
+				log.Infof("Shard master alias matches, and existing tablet agrees. Switching to master with tablet's master term start time: %v", oldTablet.MasterTermStartTime)
```

Comment:
Maybe we should also log if oldTablet.Type was not MASTER? That means the shard record and tablet record disagree, which seems like an important case to call out.

Reply:
Hmm. In this case, would it be safe to make ourselves master and match the shard record's timestamp?

Reply:
I've added handling for a couple of more corner cases. LMK what you think.